client/components/masonry-grid/index.tsx

```diff
@@ -0,0 +1,81 @@
+import clsx from 'clsx';
+import { debounce } from 'lodash';
```

Comment:
```suggestion
import { throttle } from 'lodash';
```
I think `throttle` is better than `debounce` for our use case. Imagine resizing the browser. `debounce` will keep delaying the call until we stop moving the mouse, whereas `throttle` guarantees we recalculate the layout every 100ms, meaning things don't look as broken while we resize.